src/main/java/org/spongepowered/common/data/processor/multi/entity/InvisibilityDataProcessor.java

```diff
@@ -67,13 +67,9 @@ protected boolean set(final VanishableBridge dataHolder, final Map<Key<?>, Objec
         final boolean untargetable = (Boolean) keyValues.get(Keys.VANISH_PREVENTS_TARGETING);
         final boolean vanish = (Boolean) keyValues.get(Keys.VANISH);
         dataHolder.bridge$setInvisible(invis);
-        if (vanish) {
-            dataHolder.bridge$setVanished(true);
-            dataHolder.bridge$setUncollideable(collision);
-            dataHolder.bridge$setUntargetable(untargetable);
-        } else {
-            dataHolder.bridge$setVanished(false);
-        }
+        dataHolder.bridge$setVanished(vanish);
+        dataHolder.bridge$setUncollideable(collision);
```

Comment:
So... is this right? When vanish is false, the entity in question should unquestionably be collidable. I'm a little worried that we're getting into a spiderweb here where we have a bridge method named one thing and the key named the other.

One school of thought is setting `VANISH_IGNORES_COLLISION` and `VANISH_PREVENTS_TARGETING` to `true` should not be cleared by setting `VANISH` to `false`. That feels largely expected and I err on the side of agreeing with that. I'm thus not against this change myself, but we need to make other changes to insulate ourselves against other issues.

However, I know this is in impl, but `bridge$setUncollideable` and `bridge$setUntargetable` should really be named better to indicate that they are on vanish only. Their representive `is` methods should then be renamed similarly AND include a vanish check within their bodies. I'd rather check one method for collisions, rather than tacking on a vanish check everywhere too because it's not obvious that's what needs to happen.

If we want/need separate non-vanish uncollidable/targetting states then we need to think about that more, because we now need to think about how this will work with the vanish based states.

Reply:
From what I gathered from the implementation and the API  (`This state will be ignored if the {@link Entity} is not also vanished as per {@link #VANISH}.`) the uncollideable state should be ignored in the implementation, not in the data processor.

`One school of thought is setting VANISH_IGNORES_COLLISION and VANISH_PREVENTS_TARGETING to true should not be cleared by setting VANISH to false. That feels largely expected and I err on the side of agreeing with that. I'm thus not against this change myself, but we need to make other changes to insulate ourselves against other issues.`

Agreed, but not only that. As it stands now, setting `VANISH_IGNORES_COLLISION` explicitily to `false` is ignored if you're also setting `VANISH` to `false`. That doesn't feel right to me.

`However, I know this is in impl, but bridge$setUncollideable and bridge$setUntargetable should really be named better to indicate that they are on vanish only. Their representive is methods should then be renamed similarly AND include a vanish check within their bodies. I'd rather check one method for collisions, rather than tacking on a vanish check everywhere too because it's not obvious that's what needs to happen.`
Agreed